elytron/common/src/main/java/org/wildfly/extension/elytron_common/ElytronCommonDefinitions.java

```diff
@@ -0,0 +1,121 @@
+
+        return new ResourceDefinition[]{
+                AdvancedModifiableKeyStoreDecorator.wrap(subsystemName, extensionClass,
+                        KeyStoreDefinition.configure(subsystemName, extensionClass)),
+                ModifiableKeyStoreDecorator.wrap(subsystemName, extensionClass,
+                        LdapKeyStoreDefinition.configure(subsystemName, extensionClass)),
+                ModifiableKeyStoreDecorator.wrap(subsystemName, extensionClass,
+                        FilteringKeyStoreDefinition.configure(subsystemName, extensionClass)),
+
+                SSLDefinitions.getKeyManagerDefinition(subsystemName, extensionClass),
+                SSLDefinitions.getTrustManagerDefinition(subsystemName, extensionClass),
+                SSLDefinitions.getServerSSLContextDefinition(subsystemName, extensionClass, serverOrHostController),
+                SSLDefinitions.getClientSSLContextDefinition(subsystemName, extensionClass, serverOrHostController),
+                SSLDefinitions.getServerSNISSLContextDefinition(subsystemName, extensionClass),
+
+                CertificateAuthorityDefinition.configure(subsystemName, extensionClass),
+                CertificateAuthorityAccountDefinition.configure(subsystemName, extensionClass)
+        };
+    }
+
+    public static <T> ServiceBuilder<T> commonRequirements(String subsystemName, ServiceBuilder<T> serviceBuilder) {
+        return commonRequirements(ServiceName.of(subsystemName), serviceBuilder);
+    }
+
+    public static <T> ServiceBuilder<T> commonRequirements(ServiceName subsystemName, ServiceBuilder<T> serviceBuilder) {
+        return commonRequirements(subsystemName, serviceBuilder, true, true);
+    }
+
+    public static <T> ServiceBuilder<T> commonRequirements(String subsystemName, ServiceBuilder<T> serviceBuilder, boolean dependOnProperties, boolean dependOnProviderRegistration) {
+        return commonRequirements(ServiceName.of(subsystemName), serviceBuilder, dependOnProperties, dependOnProviderRegistration);
+    }
+
+    public static <T> ServiceBuilder<T> commonRequirements(ServiceName subsystemName, ServiceBuilder<T> serviceBuilder, boolean dependOnProperties, boolean dependOnProviderRegistration) {
+        if (dependOnProperties) serviceBuilder.requires(SecurityPropertyService.serviceName(subsystemName));
+        if (dependOnProviderRegistration) serviceBuilder.requires(ProviderRegistrationService.serviceName(subsystemName));
+        return serviceBuilder;
+    }
+
+    /* Extension methods */
+
+    public static StandardResourceDescriptionResolver getResourceDescriptionResolver(final String subsystemName, Class<?> extensionClass,
+                                                                              final String... keyPrefixes) {
+
+        String resourceName = extensionClass.getPackage().getName() + ".LocalDescriptions";
+
+        StringBuilder sb = new StringBuilder(subsystemName);
```

Comment:
It seems like the subsystem name could be derived from `extensionClass.getPackage().getName()`. That might be an option that would prevent the need to pass the subsystem name everywhere.